plotly_resampler/aggregation/aggregation_interface.py

```diff
@@ -60,24 +60,61 @@ def _supports_dtype(self, s: pd.Series):
             f"{s.dtype} doesn't match with any regex in {self.dtype_regex_list}"
         )
 
-    def _get_gap_df(self, s: pd.Series) -> Optional[pd.Series]:
-        # ------- add None where there are gaps / irregularly sampled data
-        if isinstance(s.index, pd.DatetimeIndex):
-            series_index_diff = s.index.to_series().diff().dt.total_seconds()
+    @staticmethod
+    def _calc_med_diff(s: pd.Series) -> Tuple[float, np.ndarray]:
+        # ----- divide and conquer heuristic to calculate the median diff ------
+        s_idx_diff = np.diff(s.index.values)  # fyi: s_idx_diff.shape === len(s) -1
+
+        # to do so - use a quantile-based (median) approach where we reshape the data
+        # into `n_blocks` blocks and calculate the min
+        n_blcks = 128
+        if s.shape[0] > n_blcks:
+            blck_size = (s_idx_diff.shape[0]) // n_blcks
+
+            # convert the index series index diff into a reshaped view (i.e., sid_v)
+            # Remark: we start from index "1" as index "0" contains a `NaN` value
+            sid_v: np.ndarray = s_idx_diff[: blck_size * n_blcks].reshape(n_blcks, -1)
```

Comment:
NaN-bug coming from the `diff()` yielding NaN for the first element.

Reply:
aah no my bad, s_idx_diff is now a numpy.ndarray and not the original series -> so there is no nan. The code comments are not up-to-data, hence the confusion

Reply:
nice!

Reply:
no no, that remark comment above is really confusing!